crates/chia-client/src/request_map.rs

```diff
@@ -0,0 +1,65 @@
+use std::{collections::HashMap, sync::Arc};
+
+use chia_protocol::Message;
+use tokio::sync::{oneshot, Mutex, OwnedSemaphorePermit, Semaphore};
+
+#[derive(Debug)]
+pub struct Request {
+    sender: oneshot::Sender<Message>,
+    _permit: OwnedSemaphorePermit,
+}
+
+impl Request {
+    pub fn send(self, message: Message) {
+        self.sender.send(message).ok();
+    }
+}
+
+#[derive(Debug)]
+pub struct RequestMap {
+    items: Mutex<HashMap<u16, Request>>,
```

Comment:
This isn't really used as a hash map. I imagine the reason you picked this data structure was to save space in the common case of not having many messages in flight. It looks like it could also be:
```
Mutex<[Option<Request>; u16::MAX]>
```

Reply:
That seems like it could allocate a lot of memory. Most of the time you only have one or two requests pending, so this whole system is for an edge case which is unlikely to happen anyways